components/servo/site_data_manager.rs

```diff
@@ -28,31 +30,16 @@ bitflags! {
     }
 }
 
-/// Provides APIs for inspecting and managing site data.
-///
-/// `SiteDataManager` exposes information about data that is conceptually
-/// associated with a site (currently equivalent to an origin), such as
-/// web exposed storage mechanisms like `localStorage` and `sessionStorage`.
-///
-/// The manager can be used by embedders to list sites with stored data.
-/// Support for site scoped management operations (e.g. clearing data for a
-/// specific site) will be added in the future.
-///
-/// At this stage, sites correspond directly to origins. Future work may group
-/// data at a higher level (e.g. by eTLD+1).
-///
-/// Note: Network layer state (such as the HTTP cache) is intentionally not
-/// handled by `SiteDataManager`. That functionality lives in `NetworkManager`.
 #[derive(Clone, Debug, PartialEq)]
 pub struct SiteData {
     name: String,
     storage_types: StorageType,
 }
 
 impl SiteData {
-    pub fn new(name: String, storage_types: StorageType) -> SiteData {
+    pub fn new<N: Into<String>>(name: N, storage_types: StorageType) -> SiteData {
```

Comment:
```suggestion
    pub fn new(name: impl Into<String>, storage_types: StorageType) -> SiteData {
```

Let's avoid generic here.

Reply:
Yeah, that's better.